### DISCUSSION TITLE
Problem with visualization module

### DISCUSSION THREAD

**Discussion by tientrinh3008** [neperfepx/neper#512] [Q&A]
Dear All, 

As mentioned in the title, I have a problem with visualization module when running "ctest" command after build Neper. I already set "OMP_NUM_THREADS=1" but it cannot fix the issue. 

58% tests passed, 133 tests failed out of 319

Total Test time (real) = 359.85 sec

The following tests FAILED:
        185 - V/cameraangle (Failed)
        186 - V/cameracoo1 (Failed)
        187 - V/cameracoo2 (Failed)
        188 - V/cameracoo3 (Failed)
        189 - V/cameralookat1 (Failed)
        190 - V/cameralookat2 (Failed)
        191 - V/cameralookat3 (Failed)
        192 - V/cameraprojection (Failed)
        193 - V/camerasky (Failed)
        194 - V/cellcol_id (Failed)
        195 - V/cellcol_ori (Failed)
        196 - V/cellcol_real (Failed)
        197 - V/cellcol_real_colscheme (Failed)
        198 - V/cellcol_rgb (Failed)
        199 - V/cellcol_rgb2 (Failed)
        200 - V/cellscale (Failed)
        201 - V/cellscale2 (Failed)
        202 - V/cellscaletitle (Failed)
        203 - V/celltrs (Failed)
        204 - V/crystalcol_colscheme (Failed)
        205 - V/crystalcol_id (Failed)
        206 - V/crystalcol_id_cubic (Failed)
        207 - V/crystalcol_id_hexagonal (Failed)
        208 - V/crystalcol_ori (Failed)
        209 - V/crystalcol_per (Failed)
        210 - V/crystalcol_real (Failed)
        211 - V/crystalcol_real_colscheme (Failed)
        212 - V/crystalcol_rgb (Failed)
        213 - V/crystalcol_rgb2 (Failed)
        214 - V/crystaledgecol_cubic (Failed)
        215 - V/crystaledgerad_cubic (Failed)
        216 - V/crystalscale (Failed)
        217 - V/crystalscale2 (Failed)
        218 - V/crystalscaletitle (Failed)
        219 - V/crystaltrs (Failed)
        220 - V/csys (Failed)
        221 - V/csyscol (Failed)
        222 - V/csyscoo (Failed)
        223 - V/csyslabel (Failed)
        224 - V/csyslength (Failed)
        225 - V/csysrad (Failed)
        226 - V/dataelt3d (Failed)
        227 - V/dataexpr (Failed)
        228 - V/datasimtesr (Failed)
        229 - V/datasimtess (Failed)
        230 - V/ebsd1 (Failed)
        231 - V/ebsd10 (Failed)
        232 - V/ebsd11 (Failed)
        233 - V/ebsd2 (Failed)
        234 - V/ebsd3 (Failed)
        235 - V/ebsd4 (Failed)
        236 - V/ebsd5 (Failed)
        237 - V/ebsd6 (Failed)
        238 - V/ebsd7 (Failed)
        239 - V/ebsd8 (Failed)
        240 - V/ebsd9 (Failed)
        241 - V/edgecol_id (Failed)
        242 - V/edgecol_real (Failed)
        243 - V/edgecol_real_colscheme (Failed)
        244 - V/edgecol_rgb (Failed)
        245 - V/edgecol_rgb2 (Failed)
        246 - V/edgescale (Failed)
        247 - V/edgescale2 (Failed)
        248 - V/edgescaletitle (Failed)
        249 - V/edgetrs (Failed)
        250 - V/elsetcol (Failed)
        251 - V/elsetcol_ori (Failed)
        252 - V/elsetcol_real (Failed)
        253 - V/elsetcolscheme (Failed)
        254 - V/elsetscale (Failed)
        255 - V/elsetscale2 (Failed)
        256 - V/elsetscaletitle (Failed)
        257 - V/facecol_id (Failed)
        258 - V/facecol_real (Failed)
        259 - V/facecol_real_colscheme (Failed)
        260 - V/facecol_rgb (Failed)
        261 - V/facecol_rgb2 (Failed)
        262 - V/facescale (Failed)
        263 - V/facescale2 (Failed)
        264 - V/facescaletitle (Failed)
        265 - V/facetrs (Failed)
        266 - V/imageantialias (Failed)
        267 - V/imagebackground (Failed)
        269 - V/imagesize (Failed)
        271 - V/nodecol (Failed)
        272 - V/nodecol_real (Failed)
        273 - V/nodecoo (Failed)
        274 - V/nodecoo_disp (Failed)
        275 - V/nodecoofact (Failed)
        276 - V/nodescale (Failed)
        277 - V/nodescale2 (Failed)
        278 - V/polycol_id (Failed)
        279 - V/polycol_real (Failed)
        280 - V/polycol_real_colscheme (Failed)
        281 - V/polycol_rgb (Failed)
        282 - V/polycol_rgb2 (Failed)
        283 - V/polyscale (Failed)
        284 - V/polyscale2 (Failed)
        285 - V/polyscaletitle (Failed)
        286 - V/polytrs (Failed)
        287 - V/seedcol_id (Failed)
        288 - V/seedcol_per (Failed)
        289 - V/seedcol_real (Failed)
        290 - V/seedcol_real_colscheme (Failed)
        291 - V/seedcol_rgb (Failed)
        292 - V/seedcol_rgb2 (Failed)
        293 - V/seedscale (Failed)
        294 - V/seedscale2 (Failed)
        295 - V/seedscaletitle (Failed)
        296 - V/seedtrs (Failed)
        297 - V/show (Failed)
        298 - V/showelt (Failed)
        299 - V/showfaceinter (Failed)
        300 - V/showmesh1 (Failed)
        301 - V/showmesh2 (Failed)
        302 - V/slicemesh (Failed)
        303 - V/slicemesh2 (Failed)
        304 - V/slicemesh3 (Failed)
        305 - V/slicemesh4 (Failed)
        306 - V/slicemesh5 (Failed)
        307 - V/slicemesh6 (Failed)
        308 - V/tesr (Failed)
        309 - V/tesr2d (Failed)
        310 - V/vercol_id (Failed)
        311 - V/vercol_real (Failed)
        312 - V/vercol_real_colscheme (Failed)
        313 - V/vercol_rgb (Failed)
        314 - V/vercol_rgb2 (Failed)
        315 - V/verrad (Failed)
        316 - V/verscale (Failed)
        317 - V/verscale2 (Failed)
        318 - V/verscaletitle (Failed)
        319 - V/vertrs (Failed)
Errors while running CTest
Output from these tests are in: /mnt/c/CPFEM_App/neper-4.4.1/src/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.

Any suggestions or recommendations? 
Thank you.

**Comment by smallrain1996** [comment]
It is strongly recommended to update to the latest version, which is currently 4.4.2-29. Many problems can be seen to have been solved when using the latest version

**Reply by tientrinh3008** [reply]
The latest version that I can download is 4.4.1. Could you please provide the link to version 4.4.2-29?

**Reply by smallrain1996** [reply]
```
git clone https://github.com/rquey/neper.git
cd neper/src
mkdir build
cd build
cmake ..
make
sudo make install
make test
```

**Reply by rquey** [reply]
See https://neper.info/downloads.html#downloads for download instructions ([Latest version](https://neper.info/downloads.html#latest-version)).

**Comment by rquey** [comment]
Did you install Povray?
